tools/clang/unittests/HLSL/ExecutionTest.cpp

```diff
@@ -3644,6 +3689,222 @@ TEST_F(ExecutionTest, ComputeSampleTest) {
+    "#if  __SHADER_TARGET_MAJOR > 6 || (__SHADER_TARGET_MAJOR == 6 && __SHADER_TARGET_MINOR >= 7)\n"
+    "    g_texms.sample[i][id.xy] = id.x*id.y*(i+1);\n"
+    "#else\n"
+    "    g_texms[uint3(id.xy, i)] = id.x*id.y*(i+1);\n"
+    "#endif\n"
+    "  }\n"
+    "}";
+
+  static const char pCopyShader[] =
+    "#define SAMPLES 4\n"
+    "RWStructuredBuffer<float> g_out : register(u0);\n"
+    "#if  __SHADER_TARGET_MAJOR > 6 || (__SHADER_TARGET_MAJOR == 6 && __SHADER_TARGET_MINOR >= 7)\n"
+    "RWTexture2DMS<float, 4> g_texms : register(u1);\n"
+    "#else\n"
+    "RWTexture2DArray<float> g_texms : register(u1);\n"
+    "#endif\n"
+    "[NumThreads(32, 32, 1)]\n"
+    "  void main(uint3 id : SV_GroupThreadID) {\n"
+    "  for(uint i = 0; i < SAMPLES; i++) {\n"
+    "#if  __SHADER_TARGET_MAJOR > 6 || (__SHADER_TARGET_MAJOR == 6 && __SHADER_TARGET_MINOR >= 7)\n"
+    "    g_out[i*32*32 + id.y*32 + id.x] = g_texms.sample[i][id.xy];\n"
+    "#else\n"
+    "    g_out[i*32*32 + id.y*32 + id.x] = g_texms[uint3(id.xy, i)];\n"
+    "#endif\n"
+    "  }"
+    "}";
+
+  static const int NumThreadsX = 32;
+  static const int NumThreadsY = 32;
+
+#ifdef WRITEMSAA_FALLBACK
+  static const int NumSamples = 4;
+  static const int ArraySize = 4;
+#else
+  static const int NumSamples = 4;
+  static const int ArraySize = 1;
+#endif
+  static const int ThreadsPerGroup = NumThreadsX * NumThreadsY;
+  const size_t valueSize = NumSamples * ThreadsPerGroup;
+  const size_t valueSizeInBytes =  valueSize * sizeof(float);
+
+  static const int DispatchGroupX = 1;
+  static const int DispatchGroupY = 1;
+  static const int DispatchGroupZ = 1;
+
+  CComPtr<ID3D12CommandQueue> pCommandQueue;
+  CComPtr<ID3D12CommandAllocator> pCommandAllocator;
+  FenceObj FO;
+
+  CreateComputeCommandQueue(pDevice, L"WriteMSAA Queue", &pCommandQueue);
+  InitFenceObj(pDevice, &FO);
+
+  // Create root signature.
+  CComPtr<ID3D12RootSignature> pRootSignature;
+  CD3DX12_DESCRIPTOR_RANGE ranges[2];
+  ranges[0].Init(D3D12_DESCRIPTOR_RANGE_TYPE_UAV, 1, 0, 0);
+  ranges[1].Init(D3D12_DESCRIPTOR_RANGE_TYPE_UAV, 1, 1, 0);
+
+  CreateRootSignatureFromRanges(pDevice, &pRootSignature, ranges, 2);
+
+  VERIFY_SUCCEEDED(pDevice->CreateCommandAllocator(D3D12_COMMAND_LIST_TYPE_COMPUTE, IID_PPV_ARGS(&pCommandAllocator)));
+
+  // Create command list and resources
+  CComPtr<ID3D12GraphicsCommandList> pCommandList;
+  VERIFY_SUCCEEDED(pDevice->CreateCommandList(0, D3D12_COMMAND_LIST_TYPE_COMPUTE,
+                                              pCommandAllocator, nullptr, IID_PPV_ARGS(&pCommandList)));
+
+  // Set up Output Resource
+  CComPtr<ID3D12Resource> pOutputResource;
+  CComPtr<ID3D12Resource> pOutputReadBuffer;
+  CComPtr<ID3D12Resource> pOutputUploadResource;
+
+  float outVals[valueSize];
+  int ix = 0;
+  for (int i = 0; i < NumSamples; i++)
+    for (int j = 0; j < NumThreadsY; j++)
+      for (int k = 0; k < NumThreadsX; k++)
+        outVals[ix++] = (float)ix + 5;
```

Comment:
Do these values have any significance?  Could this just as well be initialized to all zeros?